### PR TITLE
Update actions to replace deprecated versions

### PR DESCRIPTION
**Pull request by elfringham**
actions/cache@v3 uses the deprecated Node.js 16 so update to use v4 which uses Node.js 20. This also applies to save and restore

**Description**

Deprecation warnings seen in logs
https://github.com/commaai/openpilot/actions/runs/8733625813

Documentation for actions/cache
https://github.com/marketplace/actions/cache

**Verification**

Code inspection

### CODE REVIEW

File: .github/workflows/auto_pr_review.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
     runs-on: ubuntu-latest
     if: github.repository == 'commaai/openpilot'
     steps:
-    - uses: Vankka/pr-target-branch-action@69ab6dd5c221de3548b3b6c4d102c1f4913d3baa
+    - uses: Vankka/pr-target-branch-action@v2.1
```
Comment (adeebshihadeh) [inline .github/workflows/auto_pr_review.yaml:27]:
```suggestion
    - uses: Vankka/pr-target-branch-action@def32ec9d93514138d6ac0132ee62e120a72aed5
```

File: .github/workflows/selfdrive_tests.yaml [hunk 2]
```diff
@@ -357,22 +357,22 @@ jobs:
...
-        uses: peter-evans/find-comment@1769778a0c5bd330272d749d12c036d65e70d39d
+        uses: peter-evans/find-comment@v3
         id: fc
         with:
           issue-number: ${{ github.event.pull_request.number }}
           body-includes: This PR makes changes to
       - name: Update comment
         if: ${{ steps.save_diff.outputs.diff != '' && env.AZURE_TOKEN != '' }}
-        uses: peter-evans/create-or-update-comment@b95e16d2859ad843a14218d1028da5b2c4cbc4b4
+        uses: peter-evans/create-or-update-comment@v4
```
Comment (adeebshihadeh) [inline .github/workflows/selfdrive_tests.yaml:367]: can you convert all these 3rd party actions back to specific commit hashes?

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

File: .github/workflows/auto_pr_review.yaml (hunk 1 above)
Comment (elfringham) [reply .github/workflows/auto_pr_review.yaml]: Done

File: .github/workflows/selfdrive_tests.yaml (hunk 2 above)
Comment (elfringham) [reply .github/workflows/selfdrive_tests.yaml]: Done

**Pull request merged**